rewards/src/lib.rs

```diff
@@ -124,11 +130,16 @@ pub mod module {
 }
 
 impl<T: Config> Pallet<T> {
-	pub fn accumulate_reward(pool: &T::PoolId, reward_currency: T::CurrencyId, reward_increment: T::Balance) {
+	pub fn accumulate_reward(
+		pool: &T::PoolId,
+		reward_currency: T::CurrencyId,
+		reward_increment: T::Balance,
+	) -> DispatchResult {
 		if reward_increment.is_zero() {
-			return;
+			return Ok(());
 		}
-		Pools::<T>::mutate_exists(pool, |maybe_pool_info| {
+		Pools::<T>::mutate_exists(pool, |maybe_pool_info| -> DispatchResult {
+			ensure!(maybe_pool_info.is_some(), Error::<T>::PoolDoesNotExist);
```

Comment:
`let Some(pool_info) = maybe_pool_info.ok_or(Error::<T>::PoolDoesNotExist)?`

Reply:
> let pool_info = maybe_pool_info.ok_or(Error::<T>::PoolDoesNotExist)?;

missed that 👌